app/scripts/modules/cloudfoundry/src/pipeline/config/validation/cfTargetImpedance.validator.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  IPipeline,
+  IStage,
+  IStageOrTriggerTypeConfig,
+  NameUtils,
+  PipelineConfigService,
+  IStageOrTriggerValidator,
+  IValidatorConfig,
+  PipelineConfigValidator,
+} from '@spinnaker/core';
+
+export interface ITargetImpedanceValidationConfig extends IValidatorConfig {
+  stageTypes?: string[];
+  stageType?: string;
+  message: string;
+}
+
+export class CfTargetImpedanceValidator implements IStageOrTriggerValidator {
+  public validate(
+    pipeline: IPipeline,
+    stage: IStage,
+    validator: ITargetImpedanceValidationConfig,
+    _config: IStageOrTriggerTypeConfig,
+  ): string {
+    const stagesToTest: IStage[] = PipelineConfigService.getAllUpstreamDependencies(pipeline, stage),
+      regions: string[] = stage['regions'] || [];
+    let allRegionsFound = true;
+
+    regions.forEach(region => {
+      let regionFound = false;
+      stagesToTest.forEach(toTest => {
+        if (toTest.type === 'deploy' && toTest['clusters'] && toTest['clusters'].length) {
```

Comment:
you can just use `toTest.clusters` here (and `stage.clusters`, `stage.credentials`, etc.)

Reply:
updated to suggested changes